lib/bloc/input/input_bloc.dart

```diff
@@ -42,8 +42,11 @@ class InputBloc extends Cubit<InputState> {
     _log.info("Tracking incoming payment: $invoiceId");
     await ServiceInjector().liquidSDK.paymentsStream.firstWhere(
       (paymentList) {
-        if (paymentList.any((e) => e.swapId == invoiceId)) {
-          if (invoiceId != null && invoiceId.isNotEmpty) {
+        if (invoiceId != null && invoiceId.isNotEmpty) {
```

Comment:
Isn't it better to subscribe to the payment event instead of the payment list?

Reply:
Done, we're listening to payment result stream now.